air_collector_node:
  ros__parameters:
    flow_rate: 28.0  # Flow rate in SCFM
    co2_intake: 1.04 # CO2 intake in mmHg
    crew_onboard: 4
    cabin_pressure: 14.7  # Cabin pressure in PSI
    temperature_cutoff: 450.0  # Temperature cutoff in Celsius
    max_crew_limit: 6
    power_consumption: 1.0  # Power in kW
    tank_capacity: 1000.0  # Air tank capacity in grams
    system_name: "demo_nova_sanctum"
    mode_of_operation: "standby"

    co2_threshold: 100.0
    moisture_threshold: 70.0
    contaminants_threshold: 30.0

    temp_kp: 0.1
    temp_ki: 0.01
    temp_kd: 0.005
    press_kp: 0.1
    press_ki: 0.01
    press_kd: 0.005


/desiccant_bed_1:
  ros__parameters:
    moisture_removal_rate: 0.20
    contaminant_removal_rate: 0.20
    emergency_threshold: 5.0

/desiccant_bed_2:
  ros__parameters:
    humidification_rate: 1.5
    emergency_threshold: 5.0

/adsorbent_bed_1:
  ros__parameters:
    co2_removal_efficiency: 0.95
    co2_to_space_ratio: 0.40
    desired_temperature: 420.0
    temperature_tolerance: 30.0
    kp: 0.6
    kd: 0.15
    co2_adsorption_rate_constant: 0.15  # Langmuir k_ads
    co2_capacity: 850.0  # Total COâ‚‚ adsorption capacity
    tank_capacity: 500.0  # Tank capacity in grams

    mode_of_operation: "standby"
/adsorbent_bed_2:
  ros__parameters:
    desorption_temperature: 400.0
    co2_desorption_rate_constant: 3.5  # Langmuir desorption rate
